lib/rules/no-tabs.js

```diff
@@ -21,25 +21,54 @@ module.exports = {
             category: "Stylistic Issues",
             recommended: false
         },
-        schema: []
+        schema: [],
+        fixable: "whitespace"
     },
 
     create(context) {
         return {
             Program(node) {
+                const source = context.getSourceCode();
+                let scriptIndex = 0;
+
                 context.getSourceLines().forEach((line, index) => {
-                    const match = regex.exec(line);
+                    let match;
+
+                    // .exec() returns null when it cannot find more matches
+                    while ((match = regex.exec(line))) {
+                        const range = [
+                            scriptIndex + match.index,
+                            scriptIndex + match.index + 1
+                        ];
+                        const astNode = source.getNodeByRangeIndex(range[0]);
 
-                    if (match) {
-                        context.report(
+                        context.report({
                             node,
-                            {
+                            loc: {
                                 line: index + 1,
                                 column: match.index + 1
                             },
-                            "Unexpected tab character."
-                        );
+                            message: "Unexpected tab character.",
+
+                            // This function needs to capture the current `range`
+                            // eslint-disable-next-line no-loop-func
```

Comment:
Because the purpose of `no-loop-func` is to prevent problematic variable references from functions inside of loops.
Now this function does not have problematic variable references.